roofit/roofitcore/inc/RooNaNPacker.h

```diff
@@ -42,6 +42,11 @@ struct RooNaNPacker {
   explicit RooNaNPacker(float value) :
     _payload(packFloatIntoNaN(value)) {  }
 
+  /// Convert into NaN-double with float packed into mantissa.
```

Comment:
I'm not a big fan of cast operators in general because it often makes the code harder to understand and prone to bugs.
For example: with the cast operator, this code would not look suspicious:
```C++
float x = rooNanPacker;
```
But you'd get some payloaded double casted to a float I guess, which is probably not intended. It the `operator double()` is replaced with some self-explaining member function, I would worry less about this.

Let me know how you think about this; if you are really attached to the cast operator this would also not be a big problem for me :)

Reply:
Ah, no. Not at all. Good point!